Tests/src/InternalTests/SpaceEntityTests.cpp

```diff
@@ -0,0 +1,1093 @@
+#include "Debug/Logging.h"
+#include "Multiplayer/NetworkEventManagerImpl.h"
+#include "Multiplayer/Script/EntityScriptBinding.h"
+#include "TestHelpers.h"
+#include "quickjspp.hpp"
+
+#include "gtest/gtest.h"
+#include <CSP/Multiplayer/Components/CustomSpaceComponent.h>
+#include <CSP/Multiplayer/Components/SplineSpaceComponent.h>
+#include <Memory/Memory.h>
+#include <PublicAPITests/SpaceSystemTestHelpers.h>
+#include <PublicAPITests/UserSystemTestHelpers.h>
+#include <atomic>
+
+using namespace csp::multiplayer;
+
+namespace
+{
+
+MultiplayerConnection* Connection;
+SpaceEntitySystem* EntitySystem;
+
+bool RequestPredicate(const csp::systems::ResultBase& Result)
+{
+	return Result.GetResultCode() != csp::systems::EResultCode::InProgress;
+}
+
+void OnUserCreated(SpaceEntity* InUser)
+{
+	EXPECT_EQ(InUser->GetComponents()->Size(), 1);
+
+	auto* AvatarComponent = InUser->GetComponent(0);
+
+	EXPECT_EQ(AvatarComponent->GetComponentType(), ComponentType::AvatarData);
+
+	std::cerr << "OnUserCreated" << std::endl;
+}
+
+} // namespace
+
+#if RUN_ALL_UNIT_TESTS || RUN_SCRIPTSYSTEM_TESTS || RUN_UPDATE_SPACE_ENTITY_GLOBAL_POSITION_TEST
+CSP_PUBLIC_TEST(CSPEngine, SpaceEntityTests, UpdateSpaceEntityGlobalPositionTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* Connection	 = SystemsManager.GetMultiplayerConnection();
+	auto* EntitySystem	 = SystemsManager.GetSpaceEntitySystem();
+
+	const char* TestSpaceName		 = "OLY-UNITTEST-SPACE-REWIND";
+	const char* TestSpaceDescription = "OLY-UNITTEST-SPACEDESC-REWIND";
+
+	char UniqueSpaceName[256];
+	SPRINTF(UniqueSpaceName, "%s-%s", TestSpaceName, GetUniqueString().c_str());
+
+	csp::common::String UserId;
+
+	// Log in
+	LogIn(UserSystem, UserId);
+
+	// Create space
+	csp::systems::Space Space;
+	CreateSpace(SpaceSystem,
+				UniqueSpaceName,
+				TestSpaceDescription,
+				csp::systems::SpaceAttributes::Private,
+				nullptr,
+				nullptr,
+				nullptr,
+				nullptr,
+				Space);
+
+	std::atomic_bool ScriptSystemReady = false;
+
+	auto ScriptSystemReadyCallback = [&ScriptSystemReady](bool Ok)
+	{
+		EXPECT_EQ(Ok, true);
+		std::cerr << "ScriptSystemReadyCallback called" << std::endl;
+		ScriptSystemReady = true;
+	};
+
+	auto [EnterResult] = AWAIT_PRE(SpaceSystem, EnterSpace, RequestPredicate, Space.Id);
+
+	EXPECT_EQ(EnterResult.GetResultCode(), csp::systems::EResultCode::Success);
+
+	EntitySystem->SetEntityCreatedCallback(
+		[](SpaceEntity* Entity)
+		{
+		});
+	EntitySystem->SetScriptSystemReadyCallback(ScriptSystemReadyCallback);
+
+	const csp::common::String& UserName = "Player 1";
+	const SpaceTransform& UserTransform
+		= {csp::common::Vector3 {1.452322f, 2.34f, 3.45f}, csp::common::Vector4 {4.1f, 5.1f, 6.1f, 7.1f}, csp::common::Vector3 {1, 1, 1}};
+	AvatarState UserAvatarState				  = AvatarState::Idle;
```

Comment:
[Question] Why are we creating an avatar here, is that required to execute scripts, I do not believe that it is. It seems to add a bit of unnecessary complexity if it isn't required.